cloud/gce: update instructions

vivekmenezes opened the pull request:
<!-- Reviewable:start -->

This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/9068)

<!-- Reviewable:end -->

vivekmenezes committed:
cloud: update instructions

Also fix some string quoting issues in terraform configs.

a-robinson commented:
Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks pending.

---

*[cloud/gce/README.md, line 15 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068#-KQfslOvo_pi4FCoa5yb:-KQfslOvo_pi4FCoa5yc:bpnb15i) ([raw file](https://github.com/cockroachdb/cockroach/blob/ebe81cb10459a057aa960ba9314da3ba86c545dc/cloud/gce/README.md#L15)):*
> ```Markdown
> 
> ## One-time setup steps
> 1. Get a `<username>@cockroachlabs.com` account.
> ```

These are our public docs for setting up a cluster on GCE, not CRL-employee-specific docs

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068)*
<!-- Sent from Reviewable.io -->

vivekmenezes commented:
Review status: 0 of 1 files reviewed at latest revision, 1 unresolved discussion.

---

*[cloud/gce/README.md, line 15 \[r1\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068#-KQfslOvo_pi4FCoa5yb:-KQfvA2JpsA41EQ6bO0N:bfwzex) ([raw file](https://github.com/cockroachdb/cockroach/blob/ebe81cb10459a057aa960ba9314da3ba86c545dc/cloud/gce/README.md#L15)):*
<details><summary><i>Previously, a-robinson (Alex Robinson) wrote…</i></summary>
> These are our public docs for setting up a cluster on GCE, not CRL-employee-specific docs
</details>
Ah ha good to know. Updated

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068)*
<!-- Sent from Reviewable.io -->

tamird commented on cloud/gce/README.md:
````diff
@@ -43,7 +45,7 @@ The following variables can be modified if necessary:
 
 ```
-$ terraform apply --var=num_instances=3
+$ terraform apply --var=num_instances=\"3\"
````
can you use single quotes here? shell nit.

vivekmenezes commented:
Review status: 0 of 1 files reviewed at latest revision, 2 unresolved discussions, some commit checks pending.

---

*[cloud/gce/README.md, line 48 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068#-KQfsIIq5erTf8d8yAtO-r2-48:-KQfxjh0Asz_uCtGC1N-:biq4r8o) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac0907df605e4304ac3fff919124c3c0093d94c2/cloud/gce/README.md#L48)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> can you use single quotes here? shell nit.
</details>
No. It's expecting a string for num_instances.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068)*
<!-- Sent from Reviewable.io -->

andreimatei commented:
LGTM

mention somewhere that terraform will create the user `ubuntu` on the instances, and associate the ssh key with that, so you need to ssh as `ubuntu@ip`

---

Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[cloud/gce/README.md, line 24 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068#-KQfwsoD9ku0YSppXIUj:-KQfwsoD9ku0YSppXIUk:b-ienis1) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac0907df605e4304ac3fff919124c3c0093d94c2/cloud/gce/README.md#L24)):*
> ```Markdown
> ```
> 5. Create a new pair of ssh keys `ssh-keygen -t rsa -C "your-email"` for GCE with location `~/.ssh/google_compute_engine`, or use an existing pair and adjust the `key_name` variable described below.
> 6. The ssh private key by default is encrypted. Decrypt the private key using: `openssl rsa -in ~/.ssh/google_compute_engine -out ~/.ssh/google_compute_engine`.
> ```

what's this about? I just copied my existing key as `google_compute_engine`, and didn't run into any issues. And I'm pretty sure I never decrypted that...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068#-:-KQg-pVMRm9HVs6BxYse:bttommi)*
<!-- Sent from Reviewable.io -->

vivekmenezes commented:
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[cloud/gce/README.md, line 24 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068#-KQfwsoD9ku0YSppXIUj:-KQg3nJoPm43_Y9BeU6L:b-7wad7) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac0907df605e4304ac3fff919124c3c0093d94c2/cloud/gce/README.md#L24)):*
<details><summary><i>Previously, andreimatei (Andrei Matei) wrote…</i></summary>
> what's this about? I just copied my existing key as `google_compute_engine`, and didn't run into any issues. And I'm pretty sure I never decrypted that...
</details>
Can you check if your private key is encrypted. You should not see 

Proc-Type: 4,ENCRYPTED

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068)*
<!-- Sent from Reviewable.io -->

vivekmenezes commented:
>  terraform will create the user ubuntu on the instances, and associate the ssh key with that, so you need to ssh as ubuntu@ip
It's already mentioned in the doc

---

Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068#-:-KQg4VETIEBw2A6sFXbS:bqux7dw)*
<!-- Sent from Reviewable.io -->

vivekmenezes commented:
Review status: 0 of 1 files reviewed at latest revision, 3 unresolved discussions, all commit checks successful.

---

*[cloud/gce/README.md, line 24 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068#-KQfwsoD9ku0YSppXIUj:-KQg59zEitAUWOE48M64:b-v0nnxp) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac0907df605e4304ac3fff919124c3c0093d94c2/cloud/gce/README.md#L24)):*
<details><summary><i>Previously, vivekmenezes wrote…</i></summary>
> Can you check if your private key is encrypted. You should not see 
> 
> Proc-Type: 4,ENCRYPTED
</details>
If you leave the passphrase blank, the key is not encrypted

I'll update the doc to say that.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068)*
<!-- Sent from Reviewable.io -->

tamird commented:
Reviewed 1 of 1 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[cloud/gce/README.md, line 24 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068#-KQfwsoD9ku0YSppXIUj:-KQg9Ut7pOmTqzYUuO9J:b-39na5c) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac0907df605e4304ac3fff919124c3c0093d94c2/cloud/gce/README.md#L24)):*
<details><summary><i>Previously, vivekmenezes wrote…</i></summary>
> If you leave the passphrase blank, the key is not encrypted
> 
> I'll update the doc to say that.
</details>
+1 that you don't need to create a new pair of keys. If you already have a key and it is encrypted, you can decrypt it: http://support.citrix.com/article/CTX122930

---

*[cloud/gce/README.md, line 48 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068#-KQfsIIq5erTf8d8yAtO-r2-48:-KQg8jbvEcLAnH7nR9fx:b-rhavns) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac0907df605e4304ac3fff919124c3c0093d94c2/cloud/gce/README.md#L48)):*
<details><summary><i>Previously, vivekmenezes wrote…</i></summary>
> No. It's expecting a string for num_instances.
</details>
Oh...this is somehow Go syntax? Odd.

This also needs to be updated in `cloud/aws/README.md`and `acceptance/terrafarm/farmer.go`

Sounds like updating those things fixes #8619.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068)*
<!-- Sent from Reviewable.io -->

vivekmenezes commented:
Review status: 0 of 3 files reviewed at latest revision, 3 unresolved discussions.

---

*[cloud/gce/README.md, line 24 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068#-KQfwsoD9ku0YSppXIUj:-KQgUyqSptuuyWPCnizT:b-7tfmjp) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac0907df605e4304ac3fff919124c3c0093d94c2/cloud/gce/README.md#L24)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> +1 that you don't need to create a new pair of keys. If you already have a key and it is encrypted, you can decrypt it: http://support.citrix.com/article/CTX122930
</details>
added

---

*[cloud/gce/README.md, line 48 \[r2\]](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068#-KQfsIIq5erTf8d8yAtO-r2-48:-KQgUxjG_fLTHkYobf-f:b-h2u1ns) ([raw file](https://github.com/cockroachdb/cockroach/blob/ac0907df605e4304ac3fff919124c3c0093d94c2/cloud/gce/README.md#L48)):*
<details><summary><i>Previously, tamird (Tamir Duberstein) wrote…</i></summary>
> Oh...this is somehow Go syntax? Odd.
> 
> This also needs to be updated in `cloud/aws/README.md`and `acceptance/terrafarm/farmer.go`
> 
> Sounds like updating those things fixes #8619.
</details>
added aws configs to this PR.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/cockroachdb/cockroach/9068)*
<!-- Sent from Reviewable.io -->
